rabbitmq-operator-model/src/main/java/com/indeed/operators/rabbitmq/model/crd/rabbitmq/PolicyDefinitionSpec.java

```diff
@@ -0,0 +1,163 @@
+    private final Long maxLengthBytes;
+    private final Long messageTtl;
+    private final String queueMasterLocator;
+
+    public PolicyDefinitionSpec(
+            @JsonProperty("alternate-exchange") final String alternateExchange,
+            @JsonProperty("dead-letter-exchange") final String deadLetterExchange,
+            @JsonProperty("dead-letter-routing-key") final String deadLetterRoutingKey,
+            @JsonProperty("expires") final Long expires,
+            @JsonProperty("ha-mode") final String haMode,
+            @JsonProperty("ha-params") final Long haParams,
+            @JsonProperty("ha-promote-on-shutdown") final String haPromoteOnShutdown,
+            @JsonProperty("ha-sync-batch-size") final Long haSyncBatchSize,
+            @JsonProperty("ha-sync-mode") final String haSyncMode,
+            @JsonProperty("max-length") final Long maxLength,
+            @JsonProperty("max-length-bytes") final Long maxLengthBytes,
+            @JsonProperty("message-ttl") final Long messageTtl,
+            @JsonProperty("queue-master-locator") final String queueMasterLocator
+    ) {
+        this.alternateExchange = alternateExchange;
+        this.deadLetterExchange = deadLetterExchange;
+        this.deadLetterRoutingKey = deadLetterRoutingKey;
+        this.expires = expires;
+        this.haMode = haMode;
+        this.haParams = haParams;
+        this.haPromoteOnShutdown = haPromoteOnShutdown;
+        this.haSyncBatchSize = haSyncBatchSize;
+        this.haSyncMode = haSyncMode;
+        this.maxLength = maxLength;
+        this.maxLengthBytes = maxLengthBytes;
+        this.messageTtl = messageTtl;
+        this.queueMasterLocator = queueMasterLocator;
+    }
+
+    public String getAlternateExchange() {
+        return alternateExchange;
+    }
+
+    public String getDeadLetterExchange() {
+        return deadLetterExchange;
+    }
+
+    public String getDeadLetterRoutingKey() {
+        return deadLetterRoutingKey;
+    }
+
+    public Long getExpires() {
+        return expires;
+    }
+
+    public String getHaMode() {
+        return haMode;
+    }
+
+    public Long getHaParams() {
+        return haParams;
+    }
+
+    public String getHaPromoteOnShutdown() {
+        return haPromoteOnShutdown;
+    }
+
+    public Long getHaSyncBatchSize() {
+        return haSyncBatchSize;
+    }
+
+    public String getHaSyncMode() {
+        return haSyncMode;
+    }
+
+    public Long getMaxLength() {
+        return maxLength;
+    }
+
+    public Long getMaxLengthBytes() {
+        return maxLengthBytes;
+    }
+
+    public Long getMessageTtl() {
+        return messageTtl;
+    }
+
+    public String getQueueMasterLocator() {
+        return queueMasterLocator;
+    }
+
+    @JsonIgnore
+    public Definition asDefinition() {
+        Definition definition = new Definition();
+
+        if (getAlternateExchange() != null) {
+            definition = definition.withAlternateExchange(getAlternateExchange());
```

Comment:
Can you pass `null` to `definition.withAlternateExchange`?  So you could avoid the null check at least?

Reply:
unfortunately it seems the "null"s get serialized in the request payload. we could probably update rabbitmq-admin to not do that. or at least provide the option.

or EVEN BETTER, make the api factory non-static and supply our own ObjectMapper.